integration/generic-tests/uefiboot_test.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build amd64,!race
+
+package integration
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	expect "github.com/google/goexpect"
+	"github.com/u-root/u-root/pkg/qemu"
+	"github.com/u-root/u-root/pkg/vmtest"
+)
+
+// TestUefiboot tests uefiboot commmands to boot to uefishell.
+func TestUefiBoot(t *testing.T) {
+	if vmtest.TestArch() != "amd64" {
+		t.Skipf("test not supported on %s", vmtest.TestArch())
+	}
+
+	payload := "UEFIPAYLOAD.fd"
+	src := fmt.Sprintf("/home/circleci/%v", payload)
+	if tk := os.Getenv("UROOT_TEST_UEFIPAYLOADL_DIR"); len(tk) > 0 {
+		src = filepath.Join(tk, payload)
+	}
+
+	if _, err := os.Stat(src); err != nil && os.IsNotExist(err) {
+		t.Skip("UEFI payload  image is not found.")
+	}
+
+	// Create the CPIO and start QEMU.
+	q, cleanup := vmtest.QEMUTest(t, &vmtest.Options{
+		TestCmds: []string{"uefiboot /dev/sda"},
+		QEMUOpts: qemu.Options{
+			Devices: []qemu.Device{
+				qemu.IDEBlockDevice{File: src},
+				qemu.ArbitraryArgs{"-machine", "q35"},
+				qemu.ArbitraryArgs{"-m", "2048"},
+			},
+		},
+	})
+	defer cleanup()
+
+	if out, err := q.ExpectBatch([]expect.Batcher{
+		&expect.BExp{R: "PROGRESS CODE: V02070003"},
```

Comment:
what prints this progress code? ideally, leave a comment in code describing it

Reply:
Done.